### PR TITLE
Log console

### PR DESCRIPTION
**Pull request by hbcarlos**
Added new examples:
* log-messages: To send messages to the log console.
* custom-log-console: To create a new log console.
* context-menu: To add buttons to a context menu #100 

Feel free to change everything and give me feedback if something is wrong : )

ping @jtpio

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks @hbcarlos! 

The CI failure on all examples is unrelated. Should be fixed by: https://github.com/jupyterlab/extension-examples/pull/102

**Comment by jtpio** [comment]
We'll also want to add the new examples to the GitHub Actions workflow here:

https://github.com/jupyterlab/extension-examples/blob/210e6c3be4eb50e6af96cfa62fc8d5ea6a0e1bd5/.github/workflows/main.yml#L16-L29

**Comment by jtpio** [comment]
Just rebased to include the recent CI fix from #102.

**Comment by jtpio** [comment]
Disabling the `@typescript-eslint/camelcase` rule for now, since the messages used in the example can be `nbformat.IOutput` objects with the `output_type` key:

```typescript
const data: nbformat.IOutput = {
  output_type: 'display_data',
  data: {
    'text/plain': 'Hello world nbformat!!'
  }
};
```

The alternative would be to add ignore statements in the example files, but that would add distraction to the reader.

**Comment by fcollonval** [comment]
> Disabling the `@typescript-eslint/camelcase` rule for now, since the messages used in the example can be `nbformat.IOutput` objects with the `output_type` key:
> 
> ```ts
> const data: nbformat.IOutput = {
>   output_type: 'display_data',
>   data: {
>     'text/plain': 'Hello world nbformat!!'
>   }
> };
> ```
> 
> The alternative would be to add ignore statements in the example files, but that would add distraction to the reader.

Thank you both for this example. One suggestion, could we set it as warning instead of disabling it?

**Comment by jtpio** [comment]
> One suggestion, could we set it as warning instead of disabling it?

We can yes. But wouldn't the warnings be displayed in the next PR diffs on GitHub?

**Comment by fcollonval** [comment]
> But wouldn't the warnings be displayed in the next PR diffs on GitHub?

You are right (I don't think GitHub skipped unmodified lines). But as there is only one case, the noise level sounds acceptable compare to the risk of seeing more non-camelCase name.

Or we could restrict the linter to changed file:
- https://gist.github.com/oroce/11282380#gistcomment-2741146 - `grep` may bring troubles on Windows
- https://github.com/grvcoelho/lint-diff

**Comment by jtpio** [comment]
> I don't think GitHub skipped unmodified lines

It shows the warning on unmodified lines yes. For example from a recent open PR in JupyterLab:

image

But we can indeed switch to `warn` and see how it goes. It's true that for now there is only one case.

File: context-menu/Untitled.ipynb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (fcollonval) [inline context-menu/Untitled.ipynb:1]: I think this file should be deleted.

**Comment by blink1073** [comment]
Testing the new auto-cancel bot for GH Actions, opening an closing a couple times.

**Comment by blink1073** [comment]
Success!

**Comment by jtpio** [comment]
Looks good to me too, thanks again @hbcarlos for the examples and @fcollonval for the review!

**Comment by jtpio** [comment]
CI failures are not related. Getting a couple of:

```
error An unexpected error occurred: "https://registry.yarnpkg.com/rxjs/-/rxjs-6.6.0.tgz: ESOCKETTIMEDOUT".
```

**Comment by fcollonval** [comment]
@hbcarlos and @jtpio thank a lot. I pushed a very small changes to align with #105 on author name. And I merge 🎉

**Pull request merged**